Documentation/api/dap/README.md

```diff
@@ -0,0 +1,107 @@
+
+See [dap.Server.handleRequest](https://github.com/go-delve/delve/search?q=handleRequest) and capabilities set in [dap.Server.onInitializeRequest](https://github.com/go-delve/delve/search?q=onInitializeRequest) for an up-to-date list of supported requests and options.
+
+## Launch and Attach Configurations
+
+In addition to the general [DAP spec](https://microsoft.github.io/debug-adapter-protocol/specification), the server supports the following implementation-specific configuration options for starting the debug session:
+   * [AttachRequestArguments](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Attach) - see [AttachConfig godoc](https://pkg.go.dev/github.com/go-delve/delve/service/dap#AttachConfig)
+   * [LaunchRequestArguments](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Launch) - see [LaunchConfig godoc](https://pkg.go.dev/github.com/go-delve/delve/service/dap#LaunchConfig)
+
+Not all of the configurations are supported by each of the two available DAP servers:
+
+<table border=1>
+<tr>
+<th>request<th>"mode":<th>`dlv dap`<th>`dlv --headless`
+</tr>
+<tr>
+<td>launch<td>"debug"<br>"test"<br>"exec"<br>...<td>supported<td>NOT supported
+<tr>
+<td>attach<td>"local"<td>supported<td>NOT supported
+</tr>
+<tr>
+<td>attach<td>"remote"<td>NOT supported<td>supported
+</tr>
+</table>
+
+## Disconnect and Shutdown
+
+### Single-Client Mode
+
+When used with `dlv dap` or `dlv --headless --accept-multiclient=false` (default), the DAP server will shut itself down at the end of the debug session, when the client sends a [disconnect request](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Disconnect). If the debuggee was launched, it will be taken down as well. If the debugee was attached to, `terminateDebuggee` option will be respected.
+
+When the program terminates, we send a [terminated event](https://microsoft.github.io/debug-adapter-protocol/specification#Events_Terminated), which is expected to trigger a [disconnect request](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Disconnect) from the client for a session and a server shutdown. The [restart request](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Restart) is not yet supported. 
+
+The server also shuts down in case of a client connection error or SIGTERM signal, but not SIGINT (Ctrl-C) signal, which gets forwarded to the foregrounded debuggee. In either case, the server takes down a launched process, but lets an attached process continue.
```

Comment:
This is wrong, signals sent to the delve process are never forwarded to the target program. I think what you mean here is that pressing Ctrl-C on the terminal window will send the signal to the target process, instead of delve.

Reply:
I still think it's wrong, it should say something like: "but pressing Ctrl-C on the terminal where a headless instance of delve is running will send SIGINT to the debuggee. Also the "either case" in the following sentence refers to "client connection error or SIGTERM" but it could be interpreted to mean "when the server shuts down or the debuggee receives SIGINT", which is wrong.

Reply:
Done